auth/auth.go

```diff
@@ -412,115 +411,100 @@ func (auth *Authenticator) AuthenticateUser(username string, password string) Us
 	return user
 }
 
-// Authenticates a user based on a JWT token string and a set of providers.  Attempts to match the
-// issuer in the token with a provider.
-// Used to authenticate a JWT token coming from an insecure source (e.g. client request)
-// If the token is validated but the user for the username defined in the subject claim doesn't exist,
-// creates the user when autoRegister=true.
-func (auth *Authenticator) AuthenticateUntrustedJWT(token string, providers OIDCProviderMap, callbackURLFunc OIDCCallbackURLFunc) (User, jose.JWT, error) {
-
-	base.Debugf(base.KeyAuth, "AuthenticateJWT called with token: %s", base.UD(token))
+// Authenticates a user based on a JWT token string and a set of providers. Attempts to match the issuer in the
+// token with a provider. Used to authenticate a JWT token coming from an insecure source (e.g. client request).
+// If the token is validated but the user for the username defined in the subject claim doesn't exist, creates
+// the user when autoRegister=true.
+func (auth *Authenticator) AuthenticateUntrustedJWT(token string, providers OIDCProviderMap, callbackURLFunc OIDCCallbackURLFunc) (User, *oidc.IDToken, error) {
+	base.Debugf(base.KeyAuth, "AuthenticateUntrustedJWT called with token: %s", base.UD(token))
 
-	// Parse JWT (needed to determine issuer/provider)
-	jwt, err := jose.ParseJWT(token)
+	// Get the ID Token from the raw JWT string; determine issuer/provider.
+	idToken, err := GetIDToken(token)
 	if err != nil {
-		base.Debugf(base.KeyAuth, "Error parsing JWT in AuthenticateJWT: %v", err)
-		return nil, jose.JWT{}, err
-	}
-
-	// Get client for issuer
-	issuer, audiences, err := GetJWTIssuer(jwt)
-	base.Debugf(base.KeyAuth, "JWT issuer: %v, audiences: %v", base.UD(issuer), base.UD(audiences))
-	if err != nil {
-		base.Debugf(base.KeyAuth, "Error getting JWT issuer: %v", err)
-		return nil, jose.JWT{}, err
+		base.Debugf(base.KeyAuth, "Error parsing JWT string to IDToken in AuthenticateUntrustedJWT: %v", err)
+		return nil, &oidc.IDToken{}, err
```

Comment:
In the error cases (here and below), is there a reason to return a pointer to an empty token, instead of just nil?